rohmu/atomic_open.py

```diff
@@ -0,0 +1,40 @@
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Any, BinaryIO, cast, ContextManager, Generator, IO, Optional, TextIO, Union
+
+import os
+
+
+def atomic_open_for_writing_text(final_path: Path, encoding: Optional[str] = None) -> ContextManager[TextIO]:
+    return cast(ContextManager[TextIO], atomic_opener(final_path, encoding=encoding, mode="w"))
+
+
+def atomic_open_for_writing_bytes(final_path: Path) -> ContextManager[BinaryIO]:
+    return cast(ContextManager[BinaryIO], atomic_opener(final_path, mode="wb"))
```

Comment:
Why do you think they're dangerous in this context? I'm using an hardcoded mode literal.

I'm checking the literal-dependent typing, but that feels much more mage to me than what I did (python open() returning object that can handle either bytes or str is one of the bad design choices in Python imho).

Reply:
They are dangerous because the cast function does verify the type of it's input at all,
so if `atomic_opener` is later updated to return a different type, the cast will hide the error.

It's only correct right now, assuming the rest of the code doesn't change,
which isn't in the spirit of relying on type to avoid errors.

(I agree about Python and the mixed bytes/str output, heritage of the pre-typing-times, and overloads are annoying and verbose to type, but users of libraries > writers of libraries).

Reply:
I don't seem to get this to actually work. I get this error:

```
rohmu/atomic_open.py:45: error: Overloaded function implementation cannot produce return type of signature 1
rohmu/atomic_open.py:45: error: Overloaded function implementation cannot produce return type of signature 2
```
Pycharm doesn't seem to complain, though, and the type chain seems correct to me.

If I need to cast file_obj from IO[Any] to the proper IO subtype, we're back to casting and I don't know whether this is useful.